hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientRatis.java

```diff
@@ -267,50 +266,115 @@ private XceiverClientReply newWatchReply(
 
   @Override
   public XceiverClientReply watchForCommit(long index)
-      throws InterruptedException, ExecutionException, TimeoutException,
-      IOException {
+      throws InterruptedException, ExecutionException {
     final long replicatedMin = getReplicatedMinCommitIndex();
     if (replicatedMin >= index) {
       return newWatchReply(index, "replicatedMin", replicatedMin);
     }
 
+//    return watchForCommitInternalOld(index);
+    return watchForCommitInternalNew(index);
+  }
+
+  /**
+   * Old implementation that handles the retry using try-catch with a new future.
+   */
+  private XceiverClientReply watchForCommitInternalOld(long index)
```

Comment:
dead code.

Reply:
This is intentionally left there for debugging. This PR is not intended for review atm.